core/lib/zksync_core/src/api_server/web3/namespaces/snapshots.rs

```diff
@@ -0,0 +1,96 @@
+use crate::api_server::web3::backend_jsonrpc::error::internal_error;
+use crate::api_server::web3::metrics::API_METRICS;
+use crate::api_server::web3::state::RpcState;
+use crate::l1_gas_price::L1GasPriceProvider;
+use zksync_types::snapshots::{AllSnapshots, SnapshotHeader, SnapshotStorageLogsChunkMetadata};
+use zksync_types::L1BatchNumber;
+use zksync_web3_decl::error::Web3Error;
+
+#[derive(Debug)]
+pub struct SnapshotsNamespace<G> {
+    pub state: RpcState<G>,
+}
+
+impl<G> Clone for SnapshotsNamespace<G> {
+    fn clone(&self) -> Self {
+        Self {
+            state: self.state.clone(),
+        }
+    }
+}
+impl<G: L1GasPriceProvider> SnapshotsNamespace<G> {
+    pub fn new(state: RpcState<G>) -> Self {
+        Self { state }
+    }
+    pub async fn get_all_snapshots_impl(&self) -> Result<AllSnapshots, Web3Error> {
```

Comment:
Nit: There should be newlines between methods / functions for readability. `rustfmt` doesn't seem to have a granular enough control over this, unfortunately (there's [`blank_lines_lower_bound`](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#blank_lines_lower_bound), but it applies to *all* items, which is an overkill).

Reply:
fixed